spec/puppet/resource_api/base_context_spec.rb

```diff
@@ -340,13 +344,28 @@ def send_log(log, msg)
     context 'when type supports feature' do
       let(:return_val) { true }
 
-      it { expect(described_class.new('supported_feature')).to be_feature_support('simple_get_filter') }
+      it { expect(described_class.new('supported_feature', attributes)).to be_feature_support('simple_get_filter') }
     end
 
     context 'when type does not support a feature' do
       let(:return_val) { false }
 
-      it { expect(described_class.new('supported_feature')).not_to be_feature_support('simple_get_filter') }
+      it { expect(described_class.new('supported_feature', attributes)).not_to be_feature_support('simple_get_filter') }
+    end
+  end
+
+  describe '.type' do
```

Comment:
Arguably this could be a stand-alone context for `...::BaseContext::Type`, only testing `Type.new(definition)`, and testing the `context.type` accessor further up.

@da-ar , you can leave it as is, if you don't think that it'll improve readability.